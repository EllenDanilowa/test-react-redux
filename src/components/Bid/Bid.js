import React, {Component} from 'react';
import PropTypes from 'prop-types';

class Bid extends Component {
  render() {
    const {item} = this.props;

    return (
      <ul>
        <li>{item.carTitle}</li>
        <li>{item.amount}</li>
        <li>{item.created}</li>
      </ul>
    );
  }
}

Bid.propTypes = {
  item: PropTypes.shape({
    id: PropTypes.string.isRequired,
    carTitle: PropTypes.string.isRequired,
    amount: PropTypes.number.string,
    created: PropTypes.string.isRequired
  })
};

export default Bid;
