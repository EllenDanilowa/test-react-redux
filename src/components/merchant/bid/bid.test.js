import React from 'react';
import renderer from 'react-test-renderer';
import Bid from './bid';

describe('Bid', () => {
  let element;
  let item;

  beforeEach(() => {
    item = {
      id: '1',
      carTitle: 'title',
      amount: 2500,
      created: 1595116800000
    };
  });

  it('renders component with defined props and formats date', () => {
    element = renderer.create(<Bid item={item}/>).toJSON();

    expect(element).toMatchSnapshot();
  });
});
