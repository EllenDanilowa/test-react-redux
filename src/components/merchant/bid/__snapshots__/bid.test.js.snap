// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bid renders component with defined props and formats date 1`] = `
<div
  className="sc-AxjAm jHPtui"
>
  <h5
    className="sc-AxirZ bSdCkA"
  >
    title
  </h5>
  <div
    className="sc-AxiKw xpHyP"
  >
    <p
      className="sc-AxhCb jKSXXN"
    >
      <b>
        Amount: 
      </b>
      2500
    </p>
    <p
      className="sc-AxhCb jKSXXN"
    >
      <b>
        Created: 
      </b>
      July 19, 2020
    </p>
  </div>
</div>
`;
